pkg/resources/row_access_policy_acceptance_test.go

```diff
@@ -32,54 +42,393 @@ func TestAcc_RowAccessPolicy(t *testing.T) {
+			"arguments": config.SetVariable(
+				config.MapVariable(map[string]config.Variable{
+					"name": config.StringVariable("A"),
+					"type": config.StringVariable("invalid-int"),
+				}),
+			),
+			"body": config.StringVariable("case when current_role() in ('ANALYST') then true else false end"),
+		}
+	}
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: acc.TestAccProtoV6ProviderFactories,
+		PreCheck:                 func() { acc.TestAccPreCheck(t) },
+		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+			tfversion.RequireAbove(tfversion.Version1_5_0),
+		},
+		Steps: []resource.TestStep{
+			{
+				ConfigDirectory: acc.ConfigurationDirectory("TestAcc_RowAccessPolicy/basic"),
+				ConfigVariables: m(),
+				ExpectError:     regexp.MustCompile(`invalid data type: invalid-int`),
+			},
+		},
+	})
+}
+
+func TestAcc_RowAccessPolicy_DataTypeAliases(t *testing.T) {
+	id := acc.TestClient().Ids.RandomSchemaObjectIdentifier()
+	resourceName := "snowflake_row_access_policy.test"
+	m := func() map[string]config.Variable {
+		return map[string]config.Variable{
+			"name":     config.StringVariable(id.Name()),
+			"database": config.StringVariable(id.DatabaseName()),
+			"schema":   config.StringVariable(id.SchemaName()),
+			"arguments": config.SetVariable(
+				config.MapVariable(map[string]config.Variable{
+					"name": config.StringVariable("A"),
+					"type": config.StringVariable("TEXT"),
+				}),
+			),
+			"body": config.StringVariable("case when current_role() in ('ANALYST') then true else false end"),
+		}
+	}
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: acc.TestAccProtoV6ProviderFactories,
+		PreCheck:                 func() { acc.TestAccPreCheck(t) },
+		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+			tfversion.RequireAbove(tfversion.Version1_5_0),
+		},
+		Steps: []resource.TestStep{
+			{
+				ConfigDirectory: acc.ConfigurationDirectory("TestAcc_RowAccessPolicy/basic"),
+				ConfigVariables: m(),
+				Check: assert.AssertThat(t, resourceassert.RowAccessPolicyResource(t, resourceName).
+					HasNameString(id.Name()),
+					assert.Check(resource.TestCheckResourceAttr(resourceName, "argument.#", "1")),
+					assert.Check(resource.TestCheckResourceAttr(resourceName, "argument.0.type", "VARCHAR")),
+				),
+			},
+		},
+	})
+}
+
+func TestAcc_view_migrateFromVersion_0_95_0(t *testing.T) {
+	id := acc.TestClient().Ids.RandomSchemaObjectIdentifier()
+	resourceName := "snowflake_row_access_policy.test"
+	body := "case when current_role() in ('ANALYST') then true else false end"
+	m := config.Variables{
+		"name":     config.StringVariable(id.Name()),
+		"database": config.StringVariable(id.DatabaseName()),
+		"schema":   config.StringVariable(id.SchemaName()),
+		"arguments": config.SetVariable(
+			config.MapVariable(map[string]config.Variable{
+				"name": config.StringVariable("A"),
+				"type": config.StringVariable("VARCHAR"),
+			}),
```

Comment:
Could you add more arguments and maybe an argument with lowercase letters to prove my point with arg names when upgrading?

Reply:
Done.